### DISCUSSION TITLE
Run vue unmounted hook

### DISCUSSION THREAD

**Discussion by ercovi** [swup/swup#854] [Help] [answered]
### Describe your problem 🧐

I'm trying to unmount (for the moment, just console log something) a vue component (Test.vue) after page transition, but it never fires. Could be I missing something from Swup, Vue or Astro? 

### What is your question? 🙋‍♀️

Does unMounted runs with using Swup/astro with vue component? 

### URL to minimal reproduction 🔗

https://github.com/ercovi/Astro-swup-vue

### Checked all these? 📚

- [X] I have read and searched the [official docs](https://swup.js.org)
- [X] I have checked [discussions](https://github.com/orgs/swup/discussions) and [existing issues](https://github.com/issues?q=is:issue+org:swup) for answers to my question
- [X] I agree to follow this project's [Code of Conduct](https://github.com/swup/.github/blob/master/CODE_OF_CONDUCT.md)
- [X] I have provided all necessary information to the best of my knowledge

**Comment by daun** [comment, marked as answer]
You could hook right before the content is replaced by swup and run your custom logic there.

`swup.hooks.before('content:replace', (visit) => console.log(visit))`

**Reply by daun** [reply]
Swup has no knowledge of any frontend frameworks on the page, so it has no way of notifying them that the content will be unloaded. To get an unload callback for Vue components to work, there would need to be a plugin that identifies and keeps track of Vue instances in the dom. Currently, no such plugin exists but it should certainly be possible if there‘s enough demand :)
